content/mqtt/index.textile

```diff
@@ -0,0 +1,95 @@
+---
+title: MQTT Adapter
+section: mqtt
+index: 0
+jump_to:
+  Help with:
+    - Using MQTT#use
+    - Decoding MQTT messages#decoding
+    - When to use MQTT#use-cases
+    - Supported features#supported
+  Security:
+    - Security#security
+    - Token authentication#token-auth
+    - SSL#ssl
+  Extra information:
+    - Adapters#adapter-info
+    - Tutorials#tutorials
+---
+
+The "Ably MQTT protocol adapter":https://www.ably.io/adapters is able to translate back and forth between "MQTT":http://mqtt.org/ and Ably's own protocol, allowing for seamless integration of any systems you may have. MQTT (MQ Telemetry Transport) is a publish/subscribe, lightweight messaging protocol designed for constrained devices and low-bandwidth networks. One of the major uses of MQTT is with IoT (Internet of Things), where these principles are key to having effective communication between various devices.
+
+h2(#use). How to use MQTT with our adapter
+
+To use the Ably MQTT protocol adapter, you'll need to ensure you correctly configure your MQTT client as follows:
+* Set the host to "mqtt.ably.io"
+* Set ssl / tls to true and the port to 8883. (If your mqtt client does not support ssl, you should instead use port 1883, but in this case we disallow api-key auth—see "SSL usage note":#ssl below)
+* Set the keep alive time to somewhere between 15 and 60 seconds. (60s will maximise battery life, 15s will maximise reponsiveness to network issues. It must not be any higher than 60s to avoid our load balancer terminating the TCP socket for inactivity)
+* If using an api key, set the username to the part of the api key before the colon, and the password to the part after the colon
+* If using a token, set the username to the token, and leave the password blank
+
+For example, in NodeJS "mqtt package":https://www.npmjs.com/package/mqtt, you'd need to specify the following:
+
+bc[nodejs]. {
+  var options = { keepAlive: 30,
+    username: 'FIRST_HALF_OF_API_KEY',
+    password: 'SECOND_HALF_OF_API_KEY',
+    port: 8883
+  };
+  var client = mqtt.connect('mqtts:mqtt.ably.io', options);
+}
+
+This will connect using TLS through MQTT to Ably.
+
+h2(#decoding). Decoding MQTT communication
+
+Any data published through or received by the MQTT adapter will be binary encoded, due to MQTT being a binary protocol. This means that you'll need to interpret the message to get the original contents out. For example, if you tried to send a string with MQTT, with JavaScript you'd need to do the following:
+
+bc[nodejs]. {
+  var decoder = new TextDecoder();
```

Comment:
Is `TextDecoder` a native object?  I must admit I am confused as I don't know what that is or where the decoder comes from.  Can you help stupid people like me understand how this works and how to find an equivalent on my platform?